### ISSUE TITLE
Editor annotations are dropped if the editor is no longer open for any participant

### ISSUE THREAD

**Issue by tobous** [saros-project/saros#810]
When closing the editor for only one of the participants, all preexisting contribution annotations are still present when the editor is re-opened. If the editor is however closed for all participants, any annotation still present in the editor is dropped.

#### Steps to reproduce
1. Alice and Bob start a session
2. Alice and Bob both open an editor for the same shared file
3. Alice and Bob both modify the editor in order to create contribution annotations for the other party.
4. Alice and Bob both close the editor.
5. Alice and/or Bob re-open the editor.

##### Expected outcome
The contribution annotations are still present as they are still valid (did not rotate our and their text section was not deleted).

##### Actual outcome
The contribution annotations are dropped.

**Comment by tobous** [comment]
@srossbach I would classify this as a bug, but from what I remember from the Eclipse internals, the editor instances are actively detached and discarded once all participants have closed the editors. So is this a wanted behavior?

**Comment by srossbach** [comment]
Yes it is a bug. The annotations have to be "refilled" when an editor is opened again and annotations are still present for that editor in the history.
